### ISSUE TITLE
About training paramters

### ISSUE THREAD

**Issue by DoriHp** [tobybreckon/fire-detection-cnn#45]
**Describe the issue or your query**
Insert a clear and concise description of what the issue is including any steps to reproduce and/or relevant script output.

**Computing Environment (please complete the following information):**
 - OS: Windows 7 Enterprise
 - Tensorflow Version:  Tensorflow-gpu 1.12
 - OpenCV Version: 4.2.0
 
 - alternative machine learning environment / toolkit (if applicable): [e.g. keras, pytorch + version]

**Additional context**
I'm trying to reproduce your result of fireNet in your paper but the highest accuracy I got from your testing set is only 91%. I have some questions:
- Did you use the same optimizer in `fireNet.py` file to produce that result? (93% as described).
- Did you use other image processing or augmentation parameters? If the answer is yes, can you provide it?
- Is it effective if I use this model to classify whether a small flame in a picture, such as a active gas stove in a kitchen? (of course the same data will appear in the training set)
Thank in advance.

**Comment by tobybreckon** [comment]
> Did you use the same optimizer in fireNet.py file to produce that result? (93% as described).

Yes, and we have independently repeated this in a later paper based on the same code base - https://arxiv.org/abs/1911.09010

Different authors, different hardware environment, same dataset, same training optimization ... same results.

There is always some randomisation in CNN training due to the random initialisation of weights and the stochastic nature of backpropagation. I note so you are using a different operating system environment to that originally used, which will of course have a a difference in the the underlying GPU software stack.

> Did you use other image processing or augmentation parameters? 

No but I note that as per the FAQ we did **not** normalise our pixel values to the range 0 to 1 and we used BGR channel ordering, the OpenCV default, rather than RGB channel ordering for the training - this will vary depending on the data loader in use.

> Is it effective if I use this model to classify whether a small flame in a picture, such as a active gas stove in a kitchen?

Yes this should be possible. I would suggest training on examples of this type of flame, potentially using transfer learning have a pretrained on the dataset provided.

Hope this helps.

**Comment by DoriHp** [comment]
I noted that you has implement a new network based on ResNet architecture, but seem that the kink download the new paper is no longer. Can you share me your research or the architecture of that network?

**Comment by tobybreckon** [comment]
Which link is that ? .. our paper shows other models outperform ResNet 
Experimental Exploration of Compact Convolutional Neural Network Architectures for Non-temporal Real-time Fire Detection (G. Samarth, N. Bhowmik, T.P. Breckon), In Proc. Int. Conf. on Machine Learning Applications, IEEE, 2019. -  https://arxiv.org/abs/1911.09010

Provisional _unsupported_ code is available in this branch - https://github.com/tobybreckon/fire-detection-cnn/tree/icmla - , without an updated README at present but the updated download scripts grab the new data and models for ..OnFireV3 / ..OnFireV4 ... I am still to find time to merge this into main.

Hope this helps.

**Comment by DoriHp** [comment]
Sorry for my mistake. I saw your work and will train 2 new model. I hope I can archive your described result.
